data/games/under_night_in_birth_exe_late_st/movespecs/vatista.json

```diff
@@ -0,0 +1,125 @@
+{
+  "name": "Vatista",
+  "movelists": [
+    [
+      "5A",
+      "2A",
+      "5B",
+      "2B",
+      "5C",
+      "2C",
+      "j.A",
+      "j.B",
+      "j.C"
+    ],
+    [
+      "5BB",
+      "2BB",
+      {
+        "name": "Leviosa",
+        "inputs": [
+          "j.A+B"
+        ]
+      }
+    ],
+    [
+      "66B",
+      "66C"
+    ],
+    [
+      {
+        "name": "Armabellum",
+        "inputs": [
+          "B+C",
+          "[B+C]"
+        ]
+      },
+      {
+        "name": "Throw",
+        "inputs": [
+          "A+D"
+        ]
+      },
+      {
+        "name": "Guard Thrust",
+        "inputs": [
+          "214D"
+        ]
+      },
+      {
+        "name": "Veil Off",
+        "inputs": [
+          "A+B+C"
+        ]
+      }
+    ],
+    [
+      {
+        "name": "Stella Lumen",
+        "inputs": [
+          "[4]6A",
+          "[4]6B",
+          "[4]6C"
+        ]
+      },
+      {
+        "name": "Luvel Angelus",
+        "inputs": [
+          "[2]8A",
+          "[2]8B",
+          "[2]8C"
+        ]
+      },
+      {
+        "name": "Mikoruseo",
+        "inputs": [
+          "[6]4A",
+          "[6]4B",
+          "[6]4C"
+        ]
+      },
+      {
+        "name": "Deus Fragmentum",
+        "inputs": [
+          "]A[",
+          "]B[",
+          "]C[",
+          "2]A[",
+          "2]B[",
+          "2]C[",
+          "j.]A[",
+          "j.]B[",
+          "j.]C["
+        ]
+      },
+      {
+        "name": "Satelus Trianglum",
+        "inputs": []
```

Comment:
This doesn't appear to be true. This works fine in expo. Am I misunderstanding?

Reply:
It fails movespec validation.

Reply:
Shouldn't it fail to actually work then? Should we create an issue for the display components to actually validate the files its given?

Reply:
Not necessarily. So much of the app data manipulations are array mappings anyways, so mechanisms won't fail if an array is empty. Parts of the app will fail if movespecs are malformed, but not everything (like empty arrays, for example).

Reply:
It's my opinion that the app should assume that the data it's handed is correct, and we ensure it's valid before we hand it to the app.

Reply:
I guess my confusion is coming from the fact that the app isn't actually having a problem with the data as it currently is. From that perspective, the movespec is an inaccurate representation of what the app requires and we should find a way to get those two aligned.

Is there a feature that I haven't seen which does/will require `inputs` not to be an empty array?

Reply:
The concept of a move without input seems meaningless. If event occurs when you do nothing, that's like an effect or result, not a move.

Reply:
/me looks at yuzu

Reply:
`Satelus Trianglum` actually is just the effect of another move colliding with a bit Vatista has previously deployed, so you're not wrong there.

Reply:
I feel like her mechanics could be represented with conditions obviating the need for an empty input "move".

Reply:
The movespec is not a representation of what the app requires, but rather a representation of of a data format for storing unambiguous and DRY move information regardless of the existence of dustup.

Reply:
To be honest, I'm not sure how to do that without massive redundancies. eg:

```diff
{
  "name": "Stella Lumen",
  "inputs": [
    "[4]6A",
+    "[4]6[A]",
    "[4]6B",
+    "[4]6[B]",
    "[4]6C",
+    "[4]6[C]"
  ],
+  "followUps": [
+    {
+      "name": "Deus Fragmentum",
+      "inputs": [
+        "]A[",
+        "]B[",
+        "]C["
+      ]
+    },
+    {
+      "name": "Satelus Trianglum",
+      "inputs": [
+        "any"
+      ],
+      "conditions": "the move collides with a previously deployed bit"
+    }
}
```

This would be repeated for every move Vatista has. I can rewrite the movelist that way but it seems like tripling its density would make it much harder to actually use.